src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,210 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public boolean add(T value) {
-        return false;
+        linkLast(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkElementIndex(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        return addAll(size, list);
+    }
+
+    private boolean addAll(int index, List<T> c) {
```

Comment:
you don't need this method 
it should work like this: 
for (T element : list) {
   add(element) 
}

Reply:
Done